internal/hcsoci/cmd.go

```diff
@@ -0,0 +1,305 @@
+package hcsoci
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/cow"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sys/windows"
+)
+
+// Cmd represents a command being prepared or run in a process host.
+type Cmd struct {
+	// Host is the process host in which to launch the process.
+	Host cow.ProcessHost
+
+	// OS is the operating system of the process host. FUTURE: Move to cow.ProcessHost.
```

Comment:
I think we should track these `FUTURE` items separately. Either as issues, or in ADO. Thoughts?

Reply:
I'm not even sure why OS is needed for this?

Reply:
Oh I see. ProcessHost needs a `OS() string` method for `env` evaluation

Reply:
IsUVM is the more annoying one. If we added it to ProcessHost we would probably want instead something like SupportsOCISpecification(), since that's what we use this to decide.

Reply:
Did you see my other comment about IsUVM? Do we really need it?